server/querier/prometheus/converters.go

```diff
@@ -74,51 +128,36 @@ func PromReaderTransToSQL(req *prompb.ReadRequest) (sql string, db string, datas
 		}
 	}
 
-	// filter
-	for _, matcher := range q.Matchers {
-		if matcher.Name == prometheusMetricsName {
-			continue
-		}
-
-		op := ""
-		switch matcher.Type {
-		case prompb.LabelMatcher_EQ:
-			op = "="
-		case prompb.LabelMatcher_NEQ:
-			op = "!="
-		case prompb.LabelMatcher_RE:
-			op = "regexp"
-		case prompb.LabelMatcher_NRE:
-			op = "not regexp"
-		default:
-			return "", "", "", fmt.Errorf("unknown match type %v", matcher.Type)
-		}
-
-		if db != "" && db != chCommon.DB_NAME_EXT_METRICS && db != chCommon.DB_NAME_DEEPFLOW_SYSTEM {
-			filters = append(filters, fmt.Sprintf("%s %s '%s'", matcher.Name, op, matcher.Value))
-		} else {
-			filters = append(filters, fmt.Sprintf("`tag.%s` %s '%s'", matcher.Name, op, matcher.Value))
-		}
-	}
-
 	if len(metrics) == 1 {
-		return "", "", "", fmt.Errorf("not support find metrics with labels")
+		return "", "", "", errors.New("not support find metrics with labels")
 	}
 
-	if db == "" || db == chCommon.DB_NAME_EXT_METRICS || db == chCommon.DB_NAME_DEEPFLOW_SYSTEM {
-		metrics = append(metrics, extMetricsTagsName)
-	} else {
-		showSql := fmt.Sprintf("SHOW tags FROM %s WHERE time >= %d AND time <= %d", table, startTime, endTime)
-		data, _ := tagdescription.GetTagDescriptions(db, table, showSql, nil)
+	// append `tag` when isShowTagStatement has been set
```

Comment:
get all available tag names for the metrics, when the query is a ShowTag statement